universal-application-tool-0.0.1/app/views/applicant/ApplicantLayout.java

```diff
@@ -36,38 +35,47 @@ public ApplicantLayout(ViewUtils viewUtils, ProfileUtils profileUtils) {
     this.profileUtils = checkNotNull(profileUtils);
   }
 
-  protected Content render(Http.Request request, Messages messages, DomContent... mainDomContents) {
-    return render(profileUtils.currentUserProfile(request), messages, mainDomContents);
+  @Override
+  public HtmlBundle getBundle() {
+    return getBundle(new HtmlBundle());
   }
-  /** Renders mainDomContents within the main tag, in the context of the applicant layout. */
-  protected Content render(
-      Optional<UatProfile> profile, Messages messages, DomContent... mainDomContents) {
-    return htmlContent(
-        headContent("Applicant layout title"),
-        body()
-            .with(renderNavBar(profile, messages))
-            .with(mainDomContents)
-            .with(viewUtils.makeLocalJsTag("main"))
-            .withClasses(ApplicantStyles.BODY_BG_COLOR));
+
+  @Override
+  public HtmlBundle getBundle(HtmlBundle bundle) {
+    return super.getBundle(bundle);
+  }
+
+  @Override
+  public Content render(HtmlBundle bundle) {
+    bundle.addBodyStyles(ApplicantStyles.BODY_BG_COLOR);
+    String currentTitle = bundle.getTitle();
+    if (currentTitle != null && !currentTitle.isEmpty()) {
+      bundle.setTitle(currentTitle + " - CiviForm");
+    }
+    return super.render(bundle);
+  }
+
+  public Content renderWithNav(Http.Request request, Messages messages, HtmlBundle bundle) {
```

Comment:
question: When do we ever want to render this layout _without_ the nav bar?

Reply:
Never that I'm aware of?

Reply:
In that case, I was wondering if we should remove this as a public method, and instead just use `render` and always add that `addHeaderContent` line, if that makes sense.

Reply:
It requires `request` and `messages` though, so it needs more context than the typical render method and I didn't want to overload render with additional data so the method is kind of self-documenting

I think I was initially going to have an `ApplicantView`class that extended `BaseHtmlView` and had this functionality, but it felt like overkill for a single method. It might be worth looking into `BaseHtmlView` and extending that as the class is pretty worthless/unused right now.

Reply:
Yeah... `BaseHtmlView` is pretty unused as a super class right now.

Hmm, OK. But I wonder if, in that case, we should somehow discourage people to call the `render` method on ApplicantLayout...  We always want the caller to use `renderWithNav`

Reply:
I could override `render` to throw an error?

Reply:
Might not be a bad temporary solution. Your call.